ci-operator/step-registry/upi/install/openstack/docs/upi-install-openstack-docs-commands.sh

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+
+set -Eeuo pipefail
+
+# Expose credentials for the openstack client
+export OS_CLIENT_CONFIG_FILE="${SHARED_DIR}/clouds.yaml"
+
+cd "$(mktemp -d)"
+
+# Expose the UPI playbooks to the script
+cp /var/lib/openshift-install/upi/*.yaml .
+
+# Expose our premade configuration file to the script
+ln -s "${SHARED_DIR}/install-config.yaml" './install-config.yaml'
+ln -fs "${SHARED_DIR}/upi-inventory.yaml" './inventory.yaml'
+
+# Extract OS_CACERT from clouds.yaml because that's how the UPI script expects it.
+OS_CACERT="$(yq ".clouds.${OS_CLOUD}.cacert" "$OS_CLIENT_CONFIG_FILE")"
+if [ -n "$OS_CACERT" ] && [ "$OS_CACERT" != "null" ]; then
+	export OS_CACERT
+else
+	unset OS_CACERT
+fi
+
+# The cluster name is used to give a unique name to the RHCOS image in Glance
+CLUSTER_NAME="$(yq -r '.metadata.name' "${SHARED_DIR}/install-config.yaml")"
+export CLUSTER_NAME
+
+# Fill install-config.yaml gaps with the default values
+openshift-install create install-config
+
+# Make the modified install-config.yaml available for debugging purposes
+python -c 'import yaml;
+import sys
+data = yaml.safe_load(open(sys.argv[1]))
+data["pullSecret"] = "redacted"
+if "proxy" in data:
+    data["proxy"] = "redacted"
+print(yaml.dump(data))
+' 'install-config.yaml' > "${ARTIFACT_DIR}/install-config.yaml"
+
+mkdir manifests
+
+# Collect the additional manifests prepared in previous steps
+while IFS= read -r -d '' item; do
+  manifest="$( basename "${item}" )"
+  cp "${item}" "manifests/${manifest##manifest_}"
+done <   <( find "${SHARED_DIR}" \( -name "manifest_*.yml" -o -name "manifest_*.yaml" \) -print0)
+
+# UPI_DOCS is the Markdown documentation for UPI. We read it and copy its scripts to UPI_SCRIPT
+UPI_DOCS='/var/lib/openshift-install/docs/install_upi.md'
+UPI_SCRIPT="${ARTIFACT_DIR}/upi-install.sh"
+
+sed '
+	# Add a bash shebang and appropriate flags at the beginning of the target script
+	1i #!/usr/bin/env bash
+	1i set -Eeuo pipefail
+	1i
+
+	# Only pick up labelled lines from the documentation file
+	1,/e2e-openstack-upi: INCLUDE START/d
+	/e2e-openstack-upi: INCLUDE END/,/e2e-openstack-upi: INCLUDE START/d
+	/e2e-openstack-upi: INCLUDE END/,$d
+
+	# shell commands can be used directly (minus the leading PS1)
+	/^```sh/,/^```/ {
+		/^```sh/d
+		s/^$ //
+		/^```/c
+	}
+
+	# python scripts are passed as heredocs to the python interpreter
+	/^```python/,/^```/ {
+		/^```python/cpython <<EOF
+		/^```/cEOF\n
+	}
+
+	# when the markdown triple-backtick-tag ends in ".json", interpret it as
+	# a path where to put that JSON file
+	/^```.*\.json$/,/^```/ {
+		s/^```\(.*\.json\)$/cat <<EOF > "\1"/
+		/^```/cEOF\n
+	}
+	' "$UPI_DOCS" > "$UPI_SCRIPT"
+
+${SHELL} "$UPI_SCRIPT"
```

Comment:
We're getting an error because the is not mapped to a real user in the container.
```
TASK [Compute resource names] **************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: KeyError: 'getpwuid(): uid not found: 1011800000'
fatal: [localhost]: FAILED! => {"msg": "Unexpected failure during module execution.", "stdout": ""}
```

I think we can workaround this with something like https://github.com/openshift/release/blob/100c539/ci-operator/step-registry/openshift-extended/test/openshift-extended-test-commands.sh#L52-L56